source/bank_01.asm

```diff
@@ -31,6 +33,38 @@ level_name_textbox_clear:
     jp nz, .loop
     ret
 
+SECTION "calc_town_map_puzzle_index", ROMX[$61FA], BANK[$01]
+calc_town_map_puzzle_index::
+    ld a, [w_town_map_level]
+    cp LEVEL_MEW
+    jr nz, .not_mew
+    ld a, PUZZLE_MEW
+    jp farcall_ret
+
+.not_mew
+    ld a, [w_town_map_y]
+    ld c, a
+    add a
+    add a
+    add c
+    ld c, a
+    ld a, [w_town_map_x]
+    add c
+    ld c, a
+    ld a, [w_town_map_level]
+    add a
+    add a
+    add a
+    add a
+    add c
+    ld c, a
+    ld a, [w_town_map_level]
+    xor $ff
+    inc a
+    add c
+    add PUZZLE_DIGLETT
```

Comment:
Likewise a `FIRST_TOWN_MAP_PUZZLE` constant for this (the previous 15 are Easy Picross)